### ISSUE TITLE
Add aspect ratio to Bluesky image embeds

### ISSUE THREAD

**Issue by JoelOtter** [snarfed/granary#843]
I've noticed that Bluesky (at least via the official clients) has recently stopped interpreting image aspect ratio for itself, relying instead on the information [provided in the record](https://docs.bsky.app/docs/advanced-guides/posts#images-embeds):

![image](https://github.com/user-attachments/assets/d36a1808-926c-4747-a84f-a4059ecc8f5d)

Annoyingly I think the only way to address this will be to download the whole image into memory (or onto disk) and get the dimensions that way (we currently [stream](https://github.com/snarfed/granary/blob/main/granary/bluesky.py#L2258) the image data). I'm happy to take this one, @snarfed do you have any opinions about this?

**Comment by snarfed** [comment]
Yes! We've been discussing this in https://github.com/snarfed/bridgy-fed/issues/1571 . I haven't prioritized it due to the in-memory image processing, but if you want to add that - maybe with https://pypi.org/project/pillow/ ? - I'm all for it!

Also sadly Bridgy Fed uses different code to generate image blobs, https://github.com/snarfed/arroba/blob/c518924d31587dc9efa367c287c9d7d0d8654eca/arroba/datastore_storage.py#L361-L363 , but on the plus side, it should be straightforward to copy whatever we do here.

**Comment by JoelOtter** [comment]
per the Bluesky docs images are max 1MB - imo we're fine to put that in memory. I think the only reliable way to ensure a max size of image downloaded is to iterate the chunks manually in the response and append them to some buffer? My understanding is we can't rely on Content-Length headers, and some image formats like JPEG don't reliably put their size information at the start of the file.

**Comment by snarfed** [comment]
Thanks for checking! We do check Content-Length on text and JSON responses, https://github.com/snarfed/webutil/blob/6384196b72a3e4768652dc45ad55f0ff8ef9100e/util.py#L1763-L1777 , but not others. I'm not too worried about a DOS though, it's not an amplification attack, an attacker has to send us as much data as we read, and Bridgy's not a big target anyway. I'd say don't worry about it for now.

**Comment by snarfed** [comment]
...oh actually you could use `FileLimiter`, it does the chunked limited stream reading you mentioned: https://github.com/snarfed/webutil/blob/6384196b72a3e4768652dc45ad55f0ff8ef9100e/util.py#L1571-L1596

**Comment by JoelOtter** [comment]
@snarfed did this roll out to bridgy? My recent post didn't have the right aspect ratio unfortunately https://bsky.app/profile/joelotter.com/post/3ld45wjrxw52w

**Comment by JoelOtter** [comment]
Here it is when run locally, assume it just hasn't rolled out yet :) https://bsky.app/profile/joelotter.com/post/3ld46btlkdp2w

**Comment by snarfed** [comment]
Yes! Sorry about that, deploying now.
